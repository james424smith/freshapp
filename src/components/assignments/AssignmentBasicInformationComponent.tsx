import React from "react";
import { StyleProp, View } from "react-native";
import Text from "../StyledText";

type Props = {
  containerStyles: StyleProp<any>;
  labelStyles: StyleProp<any>;
  textValueStyles: StyleProp<any>;
  labelValue: string;
  textValue: string | number;
};

const AssignmentBasicInformationComponent = (props: Props) => {
  const {
    containerStyles,
    textValue,
    labelStyles,
    textValueStyles,
    labelValue,
  } = props;
  return (
    <View style={containerStyles}>
      <Text style={labelStyles}>{labelValue}</Text>
      <Text style={textValueStyles}>{textValue}</Text>
    </View>
  );
};

export default AssignmentBasicInformationComponent;
