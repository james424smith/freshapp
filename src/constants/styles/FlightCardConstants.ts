export const cardHeight = "80%";
