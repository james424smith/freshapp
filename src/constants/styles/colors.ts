import { DefaultTheme, DarkTheme } from "@react-navigation/native";
// Primary colours:
const marlowBlue = "#005AA5";
const aquamarine = "#0093BB";
// Secondary colours:
const darkBlue = "#090DB3";
const activityIndicatorBlue = "#0000ff";
const blue = "#093DBD";
const green = "#09B3A5";
const darkNavy = "#2A2E43";
const grey = "#78849E";
const white = "#FFFFFF";
const lightGrey = "#f6f6f6";
const inactiveGrey = "#C5C5C5";
const marlowBlueOp = "#005AA520";
const bubbleLeft = "#099BBD";
const darkModeMain = "#101D25";
const darkModeBlueBackground = "#232D36";
const darkInputText = "#8E9397";
const turquoise = "#83E1FA";
const inactiveDocumentContainer = "#DCDCDC";
const black = "#000";
const offlineSalaryDarkColor = "#0E3142";
const moreScreenLines = "#d6d7da";
const FAQTextColor = "#555252";
const FAQInactiveColor = "#8FAABF";
const inputSearchBackgroundColor = "#00264633";
const modalColor = "#444444";
const darkModeModalColorText = "#222222";
const darkModeModalContentItem = "#333333";
const FAQSearchBarColor = "#545C63";
const lightModeBlueBackground = "#00659B";
const authDisabledButton = "#F6F6F699";
const authDarkDisableButton = "#3F494F";
const lightInputText = "#93BFD8";

export default {
  marlowBlue,
  aquamarine,
  darkBlue,
  blue,
  green,
  darkNavy,
  grey,
  white,
  lightGrey,
  inactiveGrey,
  marlowBlueOp,
  bubbleLeft,
  darkModeMain,
  darkModeBlueBackground,
  darkInputText,
  turquoise,
  inactiveDocumentContainer,
  black,
  offlineSalaryDarkColor,
  moreScreenLines,
  activityIndicatorBlue,
  FAQTextColor,
  FAQInactiveColor,
  inputSearchBackgroundColor,
  modalColor,
  darkModeModalColorText,
  darkModeModalContentItem,
  FAQSearchBarColor,
  lightModeBlueBackground,
  authDisabledButton,
  authDarkDisableButton,
  lightInputText,
};

export const dark = {
  ...DarkTheme,
  colors: {
    ...DarkTheme.colors,
    newsText: white,
    accountVerificationID: darkInputText,
    turquoiseToWhite: turquoise,
    activityIndicatorBlue: white,
    flightColorToDarkMode: turquoise,
    newsAndMarlowBottomLinksColor: darkInputText,
    createNewAccountColorText: turquoise,
    rememberPasswordBoxText: darkInputText,
    newAquaMarineColor: turquoise,
    forgotPasswordLink: white,
    signInButtonColor: aquamarine,
    DisableSignInButtonColor: authDarkDisableButton,
    colorInputText: darkModeBlueBackground,
    chatInputText: darkModeBlueBackground,
    primary: darkModeBlueBackground,
    background: darkModeMain,
    text: white,
    fromBlackToBlue: darkModeMain,
    darkColorOfTable: turquoise,
    border: aquamarine,
    newBackgroundColor: darkModeMain,
    signInPageBackground: darkModeMain,
    greenToDark: turquoise,
    fadedText: white,
    greyBorderToDark: darkModeBlueBackground,
    inactiveTouchableOpacityStyle: white,
    darkNavyToWhite: white,
    WhiteBackgroundToDarkBlue: darkModeBlueBackground,
    isOfflineSalaryColor: offlineSalaryDarkColor,
    blackAndWhite: white,
    whiteAndBlack: black,
    tabNavigatorBlueToDark: turquoise,
    chatMessageDarkMode: darkModeMain,
    chatHistoryBackground: darkModeBlueBackground,
    whiteToDarkNavy: darkNavy,
    marlowBlueToDarkMode: darkModeBlueBackground,
    FAQQuestionToDark: white,
    searchBarFAQBackground: FAQSearchBarColor,
  },
};
export const light = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    newsText: white,
    accountVerificationID: lightInputText,
    turquoiseToWhite: white,
    flightColorToDarkMode: grey,
    newsAndMarlowBottomLinksColor: lightGrey,
    rememberPasswordBoxText: grey,
    forgotPasswordLink: FAQInactiveColor,
    createNewAccountColorText: marlowBlue,
    signInButtonColor: aquamarine,
    DisableSignInButtonColor: authDisabledButton,
    primary: lightGrey,
    background: white,
    signInPageBackground: lightGrey,
    colorInputText: inputSearchBackgroundColor,
    chatInputText: white,
    newAquaMarineColor: aquamarine,
    fromBlackToBlue: marlowBlue,
    darkColorOfTable: marlowBlue,
    newBackgroundColor: marlowBlue,
    greenToDark: green,
    fadedText: grey,
    greyBorderToDark: lightGrey,
    inactiveTouchableOpacityStyle: darkNavy,
    WhiteBackgroundToDarkBlue: white,
    isOfflineSalaryColor: marlowBlueOp,
    blackAndWhite: black,
    whiteAndBlack: white,
    tabNavigatorBlueToDark: blue,
    darkNavyToWhite: darkNavy,
    activityIndicatorBlue: activityIndicatorBlue,
    chatMessageDarkMode: lightGrey,
    chatHistoryBackground: white,
    whiteToDarkNavy: white,
    forgotPasswordEmplyoeeText: grey,
    marlowBlueToDarkMode: marlowBlue,
    FAQQuestionToDark: FAQTextColor,
    searchBarFAQBackground: inputSearchBackgroundColor,
  },
};
