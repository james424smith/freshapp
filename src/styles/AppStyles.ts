import { StyleSheet } from "react-native";
import Colors from "../constants/styles/colors";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

export const trackIconSize = wp("9%");

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
  },
  headerStyle: {
    backgroundColor: Colors.lightGrey,
    borderBottomWidth: 0,
    elevation: 0,
  },
  headerStyleChat: {
    backgroundColor: Colors.white,
    borderBottomWidth: 0,
    elevation: 0,
  },
  headerLeftStyleChatRoom: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  avatarContainerStylesChatRoom: {
    zIndex: 0,
  },
  titleValueChatRoom: {
    fontSize: hp("2%"),
    fontWeight: "bold",
    marginLeft: wp("2%"),
    color: Colors.darkNavy,
  },
  avatarOnHeaderChatRoom: {
    borderWidth: 1,
    borderColor: Colors.aquamarine,
    borderRadius: 50,
    zIndex: 0,
  },
  chatTitle: {
    fontSize: hp("3.5%"),
    fontWeight: "bold",
    color: Colors.darkNavy,
    marginLeft: wp("2.5%"),
    marginTop: wp("1%"),
  },
  moreTitleContainer: {
    marginLeft: wp("5%"),
    width: "100%",
    flexDirection: "row",
  },
  faqContainer: {
    width: "100%",
    paddingLeft: wp("15%"),
    alignItems: "flex-end",
    alignSelf: "flex-end",
  },
  moreContainer: { width: "90%", justifyContent: "flex-start" },

  moreTitleValue: {
    fontSize: hp("3%"),
    fontWeight: "bold",
    color: Colors.darkNavy,
  },
  headerStyleNotification: {
    backgroundColor: Colors.white,
    borderBottomWidth: 0,
  },
  headerStyleFlights: {
    backgroundColor: Colors.marlowBlue,
    borderBottomWidth: 0,
    elevation: 0,
  },
  titleValueFlights: {
    fontSize: hp("3.2%"),
    fontWeight: "bold",
    color: "white",
    marginLeft: wp("7%"),
  },
  headerStyleAssignment: {
    shadowColor: Colors.lightGrey,
    backgroundColor: Colors.lightGrey,
    borderBottomWidth: 0,
  },
  headerTitleStyle: {
    // fontWeight: "bold",
    // textAlign: "center",
    // alignSelf: "center",
    // alignItems: "center",
    // flex: 1,
  },

  titleContainer: {
    marginLeft: wp("4%"),
  },
  titleValue: {
    fontSize: hp("3%"),
    fontWeight: "bold",
    color: Colors.darkNavy,
  },
  logoTopContainer: {
    marginLeft: wp("4.5%"),
  },
  vesselTrackerContainer: {
    marginRight: wp("4.5%"),
  },
});
