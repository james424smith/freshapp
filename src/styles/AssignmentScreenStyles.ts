import { StyleSheet, Dimensions } from "react-native";
import Colors from "../constants/styles/colors";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

export default StyleSheet.create({
  root: {
    flex: 1,
  },
  employmentOfferText: {
    color: Colors.grey,
    borderBottomWidth: 1,
    borderBottomColor: Colors.grey,
    marginRight: wp("4%"),
    fontSize: wp("3%"),
    alignSelf: "flex-end",
  },
  employmentOfferContainer: {
    marginTop: wp("5%"),
    marginBottom: wp("5%"),
  },
  outerContainer: {
    backgroundColor: Colors.lightGrey,
  },
  vesselImageStyle: {
    flex: 1,
    width: undefined,
    height: hp("26%"),
  },
  vesselDetails: {
    flexBasis: "40%",
    margin: wp("4%"),
  },
  vesselTitle: {
    fontSize: hp("3.5%"),
    fontWeight: "bold",
    marginHorizontal: wp("4%"),
    flexBasis: "40%",
    color: Colors.darkNavy,
  },
  authLoadingContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  activeTouchableOpacity: {
    borderBottomWidth: 2,
    borderBottomColor: Colors.darkNavy,
  },
  inactiveTouchableOpacity: {
    opacity: 1,
  },
  assignmentTopContainerColor: {
    backgroundColor: Colors.lightGrey,
  },
  nextAvailabilityLabel: {
    fontSize: hp("1.5%"),
    color: Colors.grey,
    textAlign: "center",
  },
  nextAvailabilityValue: {
    fontSize: hp("3.5%"),
    fontWeight: "bold",
    color: Colors.darkNavy,
    textAlign: "center",
  },
  touchableOpacityContainer: {
    backgroundColor: Colors.lightGrey,
    flexDirection: "row",
    alignContent: "center",
    justifyContent: "space-around",
  },
  assignmentTopContainerAlignment: {
    margin: wp("8%"),
    flexDirection: "row",
    alignContent: "center",
    justifyContent: "space-around",
  },
  availabilityDateContainer: {
    flexDirection: "column",
    justifyContent: "center",
  },
  //CURRENTLY NOT USED
  activeTouchableOpacityLabel: {
    fontSize: 15,
    color: Colors.darkNavy,
    paddingBottom: 5,
    fontWeight: "bold",
  },
  //CURRENTLY NOT USED
  inactiveTouchableOpacityLabel: {
    fontSize: 14,
    color: Colors.darkNavy,
    opacity: 0.2,
    fontWeight: "bold",
  },
  scrollView: {
    flex: 1,
  },
  row: {
    marginTop: wp("4.5%"),
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
    flexWrap: "wrap",
    flex: 1,
  },
  scrollViewContainer: {
    flexGrow: 1,
  },
  joiningDateItem: {
    marginTop: wp("4%"),
    margin: wp("2%"),
    flexBasis: "45%",
    height: hp("12%"),
    borderRadius: 16,
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 1,
    borderColor: Colors.aquamarine,
  },
  endOfContractItem: {
    marginTop: wp("4%"),
    margin: wp("2%"),
    flexBasis: "45%",
    height: hp("12%"),
    borderRadius: 16,
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 1,
    borderColor: Colors.green,
  },
  secondaryItem: {
    backgroundColor: Colors.lightGrey,
    marginTop: wp("2%"),
    marginBottom: wp("2%"),
    margin: wp("2%"),
    flexBasis: "45%",
    height: hp("12%"),
    borderRadius: 16,
    borderWidth: 1,
    borderColor: Colors.lightGrey,
    justifyContent: "center",
    alignItems: "center",
  },
  vesselTrackerCircle: {
    borderWidth: 2,
    borderColor: "black",
    justifyContent: "center",
    alignItems: "center",
    borderRadius:
      Math.round(
        Dimensions.get("window").width + Dimensions.get("window").height
      ) / 2,
    width: Dimensions.get("window").width * 0.08,
    height: Dimensions.get("window").width * 0.08,
  },
  vesselDetailsLabel: {
    color: Colors.grey,
    fontSize: hp("1.8%"),
  },
  vesselDetailsValue: {
    color: Colors.darkNavy,
    fontSize: hp("2%"),
    fontWeight: "bold",
  },
  textLabel: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.aquamarine,
    fontSize: hp("1.8%"),
  },
  textValue: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.aquamarine,
    fontSize: hp("2.8%"),
    fontWeight: "bold",
  },
  textLabelOfEOC: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.green,
    fontSize: hp("1.8%"),
  },
  textValueOfEOC: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.green,
    fontSize: hp("2.8%"),
    fontWeight: "bold",
  },
  secondaryTextLabel: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.grey,
    fontSize: hp("1.8%"),
  },
  secondaryTextValue: {
    justifyContent: "center",
    alignItems: "center",
    color: Colors.darkNavy,
    fontSize: hp("2%"),
    fontWeight: "bold",
    textAlign: "center",
  },
  cargoDetailsContainerAlignment: {
    paddingTop: wp("3%"),
    backgroundColor: Colors.lightGrey,
  },
  cargoDataContainer: {
    margin: wp("5%"),
  },
  cargoTitle: {
    fontSize: hp("3.5%"),
    fontWeight: "bold",
    flexBasis: "90%",
    color: Colors.darkNavy,
  },
  cargoItemLabel: {
    fontSize: hp("1.8%"),
    paddingRight: wp("3%"),
    color: Colors.grey,
  },
  cargoItemValue: {
    color: Colors.darkNavy,
    fontSize: hp("2%"),
    fontWeight: "bold",
    maxWidth: "80%",
  },
  cargoLineStyle: {
    flexDirection: "row",
  },
  avatarStyle: {
    borderWidth: 1.5,
    borderColor: "white",
    borderRadius: 40,
  },
  avatarViewStyle: {
    borderWidth: 1.5,
    borderColor: "black",
    borderRadius: 40,
  },
  columnsContainer: {
    flex: 1,
    flexDirection: "row",
    marginBottom: wp("5.5%"),
  },
  firstContainerWidth: {
    width: "55%",
  },
  secondContainerWidth: {
    width: "45%",
  },
  lineStyle: {
    borderBottomColor: Colors.grey,
    borderBottomWidth: 0.3,
    marginTop: wp("3%"),
    marginBottom: wp("6%"),
    marginLeft: wp("5%"),
  },
  vesselDetailsContainer: {
    marginTop: wp("10%"),
  },
});
