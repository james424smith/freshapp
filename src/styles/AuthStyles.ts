import Layout from "../constants/styles/Layout";
import { StyleSheet } from "react-native";
import Colors from "../constants/styles/colors";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

export const mainIconSize = wp("7%");
export const logoWidthSize = hp("30%");
export const logoHeightSize = hp("30%");
const spaceBetween = "space-between";

export default StyleSheet.create({
  scrollViewContainer: {
    flexGrow: 1,
  },
  cancelLinkContainer: {
    paddingTop: hp("4%"),
  },
  container: {
    flex: 1,
    alignItems: "center",
  },
  centerInputText: {
    marginTop: -5,
  },
  passwordValidationWrapper: {
    flex: 0.2,
  },
  formContainer: {
    flex: 1,
    justifyContent: "space-around",
    alignItems: "center",
  },
  bgImage: {
    flex: 1,
    top: 0,
    left: 0,
    width: Layout.window.width,
    height: Layout.window.height,
    justifyContent: "center",
    alignItems: "center",
  },
  loginView: {
    marginTop: wp("20%"),
    flex: 1,
  },
  loginTitle: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  signUpView: {
    flex: 1,
    justifyContent: spaceBetween,
  },
  signUpTitle: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  signUpContainer: {
    flex: 1,
    backgroundColor: Colors.lightGrey,
    width: Layout.window.width,
  },
  signUpText: {
    fontSize: hp("4%"),
    fontWeight: "700",
    color: Colors.white,
  },
  forgotPasswordTitle: {
    fontSize: hp("4%"),
    fontWeight: "700",
    color: Colors.white,
  },
  resetPasswordText: {
    fontSize: hp("4%"),
    fontWeight: "700",
    color: "white",
  },
  enterEeIdText: {
    fontSize: hp("3.2%"),
    fontWeight: "400",
    alignSelf: "center",
    color: "black",
  },
  passwordTitle: {
    fontSize: hp("3.2%"),
    fontWeight: "400",
    alignSelf: "center",
    color: "black",
  },
  loginInput: {
    flex: 1,
    justifyContent: "center",
    marginHorizontal: wp("8%"),
  },
  newPasswordInput: {
    justifyContent: "center",
    marginTop: hp("10%"),
    flex: 1,
  },
  firstPasswordInput: {
    justifyContent: "center",
  },
  employeeIdInputContainer: {
    flex: 1,
    justifyContent: "center",
    marginTop: hp("-50%"),
  },
  passwordsInputContainer: {
    flex: 1,
    justifyContent: "center",
    marginBottom: wp("50%"),
  },
  signUpInput: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-evenly",
  },
  linearGradient: {
    flex: 1,
    width: Layout.window.width,
  },
  buttonText: {
    fontSize: 18,
    fontFamily: "Gill Sans",
    textAlign: "center",
    margin: 10,
    color: "#ffffff",
    backgroundColor: "transparent",
  },
  forgotPasswordLayout: {
    flex: 1,
    justifyContent: spaceBetween,
  },
  skipContainer: {
    alignItems: "center",
    alignSelf: "flex-end",
    paddingTop: hp("3%"),
  },
  skipButton: {
    justifyContent: spaceBetween,
    alignItems: "flex-end",
    padding: wp("3%"),
  },
  resetPasswordLayout: {
    flex: 1,
    justifyContent: spaceBetween,
  },
  belowSignInView: {
    marginTop: wp("6%"),
    marginBottom: wp("25%"),
  },
  registerText: {
    color: Colors.grey,
    fontSize: hp("1.7%"),
  },
  privacyPolicyView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  privacyPolicyLink: {
    color: Colors.grey,
    textDecorationLine: "underline",
    fontSize: hp("1.7%"),
  },
  termsConditionsView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  forgotPasswordTermsConditionsContainer: {
    paddingBottom: hp("3%"),
  },
  termsConditionsLink: {
    color: Colors.grey,
    textDecorationLine: "underline",
    fontSize: hp("1.7%"),
  },
  checkBoxContainerStyle: {
    borderColor: "transparent",
    backgroundColor: "transparent",
  },
  andBetweenStyle: {
    color: Colors.grey,
    fontSize: hp("1.7%"),
  },
  conditionsPrivacyPolicyLinks: {
    flex: 1,
  },
  newsLinkView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  newsLink: {
    color: Colors.grey,
    textDecorationLine: "underline",
  },
  imprintLinkContainer: {
    flex: 1,
  },
  imprintLinkView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  imprintLink: {
    color: Colors.grey,
    textDecorationLine: "underline",
  },
  bottomLink: {
    fontSize: hp("1.8%"),
    color: "black",
    textAlign: "center",
  },
  bottomButtonArea: {
    width: "50%",
  },
  touchableOpacityStyle: {
    width: "100%",
    height: hp("12.5%"),
    backgroundColor: Colors.lightGrey,
    alignItems: "center",
    justifyContent: "center",
    borderColor: Colors.grey,
    borderWidth: 0.5,
  },
  footerStyle: {
    position: "absolute",
    bottom: 0,
    width: Layout.window.width,
    height: hp("6.5%"),
  },
  footerContainer: {
    flex: 1,
    flexDirection: "row",
    position: "absolute",
    bottom: 0,
    alignSelf: "flex-end",
  },
  scrollViewStyle: {
    flex: 1,
  },
  scrollViewContainerStyles: {
    flex: 1,
  },
  enterEeIdTitleContainer: {
    marginTop: wp("20%"),
  },
  logoContainer: {
    alignItems: "center",
    alignSelf: "center",
    marginVertical: hp("10%"),
  },
  logoText: { fontSize: hp("4%"), color: Colors.white, fontWeight: "bold" },
  signUpTitleContainer: {
    alignItems: "center",
    alignSelf: "center",
  },
  forgotPasswordTitleContainer: {
    alignItems: "center",
    alignSelf: "center",
  },
  resetForgotPasswordTitleContainer: {
    alignItems: "center",
    alignSelf: "center",
    marginTop: hp("1%"),
  },
  credentialsContainer: {
    flex: 1,
    marginHorizontal: wp("8%"),
  },
  signUpCredentialsContainer: {
    marginHorizontal: wp("8%"),
    marginBottom: wp("6%"),
  },
  forgotPasswordCredentialsContainer: {
    justifyContent: "center",
    marginHorizontal: wp("10%"),
  },
  forgotPasswordEmployeeText: {
    color: Colors.white,
    marginLeft: wp("3%"),
    marginBottom: wp("2%"),
    fontSize: hp("1.8%"),
  },
  accountVerificationEmployeeText: {
    marginLeft: wp("4"),
    marginBottom: wp(".5%"),
    fontSize: hp("1.8%"),
  },
  accountVerificationPasswordText: {
    color: Colors.white,
    marginLeft: wp("4"),
    fontSize: hp("1.7%"),
  },
  inputBoxStyle: {
    backgroundColor: Colors.inputSearchBackgroundColor,
    borderRadius: 30,
    height: hp("6.5%"),
    alignItems: "center",
    marginBottom: wp("3%"),
  },
  inputBoxStyleForgotPassword: {
    backgroundColor: Colors.white,
    borderRadius: 30,
    height: hp("6.5%"),
    alignItems: "center",
    marginVertical: wp("3%"),
  },
  inputValueStyle: {
    fontSize: hp("2.2%"),
    marginLeft: wp("3%"),
    color: "black",
    fontFamily: "Roboto-Regular",
    flexDirection: undefined,
    width: Layout.window.width * 0.8,
  },
  dateOfBirthInputStyles: {
    fontSize: hp("2.2%"),
    marginLeft: wp("3%"),
    color: "black",
    width: Layout.window.width * 0.8,
    opacity: 0.5,
  },
  disableOpacity: {
    opacity: 1,
  },
  labelValueStyle: {
    fontSize: hp("2.2%"),
    marginLeft: wp("3%"),
    color: "black",
    width: Layout.window.width * 0.8,
  },
  removeUnderline: {
    borderBottomWidth: 0,
    zIndex: 1,
  },
  signInButton: {
    height: hp("6.7%"),
    backgroundColor: Colors.marlowBlue,
    borderRadius: 30,
    marginHorizontal: wp("10%"),
    marginBottom: hp("2%"),
  },
  signUpButton: {
    height: hp("7.5%"),
    backgroundColor: Colors.marlowBlue,
    borderRadius: 25,
    marginBottom: hp("2%"),
  },
  signInButtonView: {
    marginVertical: wp("6%"),
  },
  signUpButtonView: {
    width: "100%",
    marginVertical: wp("6%"),
  },
  textOnButton: {
    fontSize: hp("2.3%"),
    color: Colors.white,
  },
  signUpLink: {
    color: Colors.white,
    fontSize: hp("2%"),
    fontWeight: "bold",
  },
  cancelLink: {
    fontSize: hp("2%"),
    color: Colors.darkNavy,
  },
  forgotPasswordLink: {
    color: Colors.grey,
    fontSize: hp("2%"),
  },
  linkStyle: {
    backgroundColor: "transparent",
  },
  spaceContainer: {
    marginBottom: wp("18%"),
  },
  dateIconStyle: {
    position: "absolute",
    right: wp("2%"),
    marginRight: 0,
  },
  dateInputStyle: {
    alignItems: "flex-start",
    borderWidth: 0,
    borderBottomWidth: 1,
    borderBottomStartRadius: -15,
    marginLeft: wp("3%"),
    width: Layout.window.width * 0.8,
  },
  placeholderStyle: {
    color: Colors.grey,
    fontSize: hp("2.2%"),
    marginLeft: wp("3%"),
  },
  dateTextStyle: {
    fontSize: hp("2%"),
    marginLeft: wp("3%"),
  },
  datePickerStyle: {
    backgroundColor: Colors.white,
    borderRadius: 30,
    width: wp("80%"),
    height: hp("6.5%"),
    alignItems: "center",
  },
  termsOnFooter: {
    flex: 1,
  },
  navigationIconsCentered: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginBottom: wp("4%"),
  },
  passwordMessageContainer: {
    alignItems: "center",
  },
  passwordValidationSection: {
    flex: 1,
  },
  passwordValidationMapChecks: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    flexWrap: "wrap",
    justifyContent: "center",
  },

  passwordValidationContainer: {
    flex: 1,
    flexDirection: "row",
    width: wp("80%"),
    flexWrap: "wrap",
    justifyContent: spaceBetween,
  },
  passwordValidationCheckIcon: {
    width: wp("10%"),
    alignItems: "center",
  },

  passwordValidationCheckedIconText: {
    color: Colors.green,
    width: wp("70%"),
    alignSelf: "center",
  },
  passwordValidationUncheckedIconText: {
    color: Colors.grey,
    width: wp("70%"),
    alignSelf: "center",
  },

  passwordMainMessage: {
    fontSize: hp("1.8%"),
    color: Colors.grey,
    fontWeight: "bold",
  },
  passwordBulletMessage: {
    fontSize: hp("1.8%"),
    color: Colors.grey,
  },
  signUpLoginView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  signUpLoginText: {
    fontSize: hp("1.9%"),
    color: Colors.white,
  },
  signUpLoginLink: {
    color: Colors.white,
    textDecorationLine: "underline",
    textDecorationColor: Colors.white,
    fontSize: hp("1.9%"),
    fontWeight: "bold",
  },
  goBackContainer: {
    alignItems: "flex-start",
    paddingTop: hp("5%"),
  },
  exitButtonContainer: { alignItems: "flex-end", paddingTop: hp("2.5%") },
  errorMessageContainer: { paddingLeft: wp("5%") },
  errorMessageText: { color: Colors.white, fontSize: hp("1.5%") },
});
