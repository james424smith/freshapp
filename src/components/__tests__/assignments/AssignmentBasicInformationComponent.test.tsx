import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import AssignmentBasicInformationComponent from "../../assignments/AssignmentBasicInformationComponent";
import styles from "../../../styles/AssignmentScreenStyles";
import { useTheme } from "@react-navigation/native";

describe("AssignmentBasicInformationComponent snapshot test", () => {
  const theme = useTheme();
  const containerStyles = {
    ...styles.joiningDateItem,
    borderColor: theme.colors.newAquaMarineColor,
  };
  const labelStyles = {
    ...styles.textLabel,
    color: theme.colors.newAquaMarineColor,
  };
  const textValueStyles = {
    ...styles.textValue,
    color: theme.colors.newAquaMarineColor,
  };

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render AssignmentBasicInformationComponent correctly as per the ui made ", () => {
    const wrapper = render(
      <AssignmentBasicInformationComponent
        containerStyles={containerStyles}
        labelStyles={labelStyles}
        textValueStyles={textValueStyles}
        labelValue={"Some Label"}
        textValue={"Some Value"}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
