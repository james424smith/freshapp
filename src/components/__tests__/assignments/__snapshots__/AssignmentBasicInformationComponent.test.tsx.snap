// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentBasicInformationComponent snapshot test should render AssignmentBasicInformationComponent correctly as per the ui made  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": undefined,
      "borderRadius": 16,
      "borderWidth": 1,
      "flexBasis": "45%",
      "height": 160,
      "justifyContent": "center",
      "margin": 15,
      "marginTop": 30,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "alignItems": "center",
          "color": undefined,
          "fontSize": 24,
          "justifyContent": "center",
        },
      ]
    }
  >
    Some Label
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "alignItems": "center",
          "color": undefined,
          "fontSize": 37.5,
          "fontWeight": "bold",
          "justifyContent": "center",
        },
      ]
    }
  >
    Some Value
  </Text>
</View>
`;
